migrations/statictypes/statictype_migration.go

```diff
@@ -154,7 +165,30 @@ func (m *StaticTypeMigration) Migrate(
 	return
 }
 
-func (m *StaticTypeMigration) maybeConvertStaticType(staticType, parentType interpreter.StaticType) interpreter.StaticType {
+func (m *StaticTypeMigration) maybeConvertStaticType(
+	staticType interpreter.StaticType,
+	parentType interpreter.StaticType,
+) (
+	resultType interpreter.StaticType,
+) {
+	// Consult the cache and cache the result
+	// at the root of the migration,
+	// i.e. when the parent type is nil.
+
+	if parentType == nil {
```

Comment:
Any particular reason to read from the cache only if the type is the root? If there are two types, say `[T]` and `{String: T}`, isn't it OK/safe to read `T` from the cache?

Reply:
Initially I didn't have this filter in place and got confused why the tests failed.

The reason is that the intersection type migration depends on the parent type, `{Ts}` in `&{Ts}` is migrated differently from `{Ts}`.

I'll extend the comment with that context.